src/bridging/BPJSKompilasiBerkasKlaim.java

```diff
@@ -3673,42 +3706,52 @@ private void gabung() {
         // exportSKDP("009");
         // exportSPRI("010");
         if (exportSukses) {
-            mergePDF();
+            exportSukses = mergePDF();
+        } else {
+            JOptionPane.showMessageDialog(rootPane, "Tidak bisa mengekspor sebagai PDF!");
+        }
+        
+        if (exportSukses) {
             cleanupAfterMerge(true);
             JOptionPane.showMessageDialog(rootPane, "Export PDF berhasil!");
         } else {
-            JOptionPane.showMessageDialog(rootPane, "Tidak bisa mengekspor sebagai PDF!");
             exportSukses = true;
         }
     }
 
-    private void mergePDF() {
+    private boolean mergePDF() {
+        boolean sukses = true;
         PDFMergerUtility pdfMerger = new PDFMergerUtility();
         File folder = new File("./berkaspdf/" + tanggalExport);
         File[] files = folder.listFiles();
         if (files != null) {
-            Arrays.sort(files, (file1, file2) -> file1.getName().compareTo(file2.getName()));
-            for (File file : files) {
-                if (file.isFile() && file.getName().endsWith(".pdf") && file.getName().startsWith(lblNoSEP.getText() + "_")) {
-                    try {
-                        pdfMerger.addSource(file);
-                    } catch (IOException e) {
-                        System.err.println("Error adding file: " + file.getName());
+            try {
+                Arrays.sort(files, (file1, file2) -> file1.getName().compareTo(file2.getName()));
+                for (File file : files) {
+                    if (file.isFile() && file.getName().endsWith(".pdf") && file.getName().startsWith(lblNoSEP.getText() + "_")) {
+                        try {
+                            pdfMerger.addSource(file);
+                        } catch (Exception e) {
+                            System.err.println("Error adding file: " + file.getName());
+                            sukses = false;
+                            throw e;
+                        }
                     }
                 }
-            }
-            pdfMerger.setDestinationFileName("./berkaspdf/" + tanggalExport + "/" + lblNoSEP.getText() + ".pdf");
-            try {
+                pdfMerger.setDestinationFileName("./berkaspdf/" + tanggalExport + "/" + lblNoSEP.getText() + ".pdf");
                 pdfMerger.mergeDocuments(MemoryUsageSetting.setupMainMemoryOnly());
```

Comment:
Mungkin strategi memory usage bisa diganti dengan [ini].(https://pdfbox.apache.org/docs/2.0.0/javadocs/org/apache/pdfbox/io/MemoryUsageSetting.html#setupTempFileOnly())
```suggestion
                pdfMerger.mergeDocuments(MemoryUsageSetting.setupTempFileOnly(50 * 1_000_000));
```